CONTRIBUTING.md

```diff
@@ -27,6 +47,67 @@ pull request reviewed, the reviewer(s) may ask you to complete additional
 work, tests, or other changes before your pull request can be ultimately
 accepted.
 
+# Maintaining Dart Frog
+
+## Setting up your local development environment
+
+Prerequisites:
+
+- Install a valid [Dart SDK](https://dart.dev/get-dart) in your local environment, it should be compatible with the latest version of [Dart Frog CLI](https://github.com/VeryGoodOpenSource/dart_frog/blob/main/packages/dart_frog_cli/pubspec.yaml). If you have Flutter installed, you likely have a valid Dart SDK version already installed.
+- [Mason CLI][mason_install_link] (to run and test the `bricks`);
+- [Node.js][node_js_dowload_link] (For the VS Code extension and the docs site);
+- Capability to run shell scripts (for the scripts under `tool/`).
+
+## Understanding the `packages/` contents:
+
+### `dart_frog`
+
+This is the user-facing package of the Dart Frog SDK, which means that Dart Frog users will be using its API to construct servers and runtime operations. It contains logic for request parsing, middleware, and response creation. 
+
+### `dart_frog_gen`
+
+This is the internal package used by the Dart Frog tooling to interpret the file disposition and from it construct a Dart Frog server. 
+
+> :warning: **Warning**: this package is a dependency on the bricks bundled into the CLI. This means that any changes that break the bricks should be released with a major version, otherwise dart frog users may be blocked from performing tasks such as `dev`, `build`, and `new`.
+
+### `dart_frog_cli`
+
+A Dart command line interface package that serves as the main tool for Dart Frog. It includes bundled versions of the bricks under `bricks/`. To sync the source code of the bricks with new bundles, run `tool/generate_bundles.sh`.
+
+
+### Companion packages
+
+The other items under `packages/` are companion packages in which dart_frog users may include on their project for specific server-side capabilities, such as auth (`dart_frog_auth`) and WebSockets (`dart_frog_web_socket`)
+
+
+## Releasing versions of packages
+
+Before starting the release process of an individual package, first check:
+
+- If your local `main` branch is up to date;
+- If the pipeline is green for the given package;
+- How this release will possibly affect other packages. Make sure to open follow-up PRs updating this package usage in other Dart Frog packages that depend on it, after the release.
+
+Checkout on `main` and run the script under `./tool/release_ready.sh` in the package root repository with the desired new version. 
+
+Example:
+```shell
+# on packages/dart_frog, release dart frog v2 (lol)
+../../tool/release_ready.sh 2.0.0 
+```
+
+This will generate a Changelog and create a release branch. Review the changelog, make the desired changes, commit, push, and open a PR. 
+After the PR is merged, create a [release on GitHub][github_release_link]. A Workflow should take care of publishing the new version on pub.dev.
```

Comment:
What do you think of providing a step by step list?

```suggestion
Before starting the release process of an individual package, first check:

1. If your local `main` branch is up to date:
```shell
# ☁️ Ensure you're up to date with the GitHub remote
git checkout main
git fetch
git status
/```

2. Ensure the [GitHub pipeline](https://github.com/VeryGoodOpenSource/dart_frog/actions) is green (has passed successfuly) for your given package.

3. Checkout on `main` and run the script under `./tool/release_ready.sh` in the package root repository with the desired new version. 

```shell
# Run the release ready script (from packages/dart_frog)
../../tool/release_ready.sh 2.0.0 
/```

This will update the version of dart_frog to 2.0.0, update the dart_frog CHANGELOG.md, create and checkout to a local release branch.

4. Review the recently updated CHANGELOG file. You should manually amend the content were necessary. For example, by removing the redundant scope of some semantic pull requests or removing superfluous or unrelated logged changes.

5. Commit, push and a pull request from the new release branch.

6. Once merged, create a [release on GitHub][github_release_link]. A Workflow should take care of publishing the new version on the appropriate package manager.

7. Open follow-up pull requests updating this package usage in any other Dart Frog package that depends on this new release.
```

> Note, I've used /``` to terminate code blocks since I couldn't find a way to avoid terminating the suggestion block.